### PR TITLE
[Added] Gateway Cluster Cache option

### PR DESCRIPTION
**Pull request by idanmantin**
Add support for cluster cache

### CODE REVIEW

File: charts/akeyless-api-gateway/values.yaml [hunk 1]
```diff
@@ -335,6 +335,12 @@ cachingConf:
   ## The amount of time during which a secret should be cached in minutes
   # cacheTTL: 60
 
+  useClusterCache: true
+  clusterCacheEncryptionKey: #"R+jBj+VqBno5hM1PR3bGMYw5enpM6mmrJ8IYDqrbvq4="
+  # Specifies an existing secret for the cluster cache configuration, must include:
+  # - cluster-cache-encryption-key
+  clusterCacheEncryptionKeyExistingSecret: belbal
```
Comment (renanaAkeyless) [inline charts/akeyless-api-gateway/values.yaml:342]: remove the belbal

File: charts/akeyless-api-gateway/values.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ cache:
 image:
   repository: akeyless/base
   pullPolicy: Always
-  tag: latest
+  tag: 1.0.0
```
Comment (renanaAkeyless) [inline charts/akeyless-api-gateway/values.yaml:22]: remove 1.0.0

File: charts/akeyless-api-gateway/templates/_helpers.tpl [hunk 3]
```diff
@@ -108,6 +108,14 @@ Generate chart secret name
         {{ $.Release.Name }}-cache-secret
 {{- end -}}
 
+{{- define "akeyless-api-gw.clusterCacheEncryptionKeyExist" -}}
+        {{- if .Values.cachingConf.clusterCacheEncryptionKey -}}
+            {{ include "akeyless-api-gw.secretName" . }}
```
Comment (renanaAkeyless) [inline charts/akeyless-api-gateway/templates/_helpers.tpl:113]: {{ include "akeyless-api-gw.secretName" . }} ?? this is not related..
please check

File: charts/akeyless-api-gateway/values.yaml [hunk 4]
```diff
@@ -335,6 +335,12 @@ cachingConf:
   ## The amount of time during which a secret should be cached in minutes
   # cacheTTL: 60
 
+  useClusterCache: true
+  clusterCacheEncryptionKey: #"R+jBj+VqBno5hM1PR3bGMYw5enpM6mmrJ8IYDqrbvq4="
```
[the remark this answers is not in the document]
Comment (idanmantin) [reply charts/akeyless-api-gateway/values.yaml:339]: Removed that.

File: charts/akeyless-api-gateway/values.yaml (hunk 1 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/values.yaml:342]: Done

File: charts/akeyless-api-gateway/values.yaml (hunk 2 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/values.yaml:22]: Done

File: charts/akeyless-api-gateway/values.yaml [hunk 5]
```diff
@@ -353,6 +353,11 @@ cachingConf:
   ## The amount of time during which a secret should be cached in minutes
   # cacheTTL: 60
 
+  useClusterCache: true
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/values.yaml:356]: default should be true?

File: charts/akeyless-api-gateway/README.md [hunk 6]
```diff
@@ -114,14 +114,15 @@ The following table lists the configurable parameters of the API Gateway chart a
...
+| `cachingConf.proActiveCaching.minimumFetchingTime`    | When using Caching or/and Proactive Caching, additional secrets will be fetched upon requesting a secret, based on the requestor's access policy. Define minimum fetching interval to avoid over fetching in a given time frame. name | `nil`   |
+| `cachingConf.proActiveCaching.dumpInterval`           | To ensure service continuity in case of power cycle and network outage secrets will be backed up periodically per backup interval.                                                                                                    | `nil`   |
+| `cachingConf.useClusterCache`                         | Should cluster caching be enabled                                                                                                                                                                                                     | `false` |
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/README.md:124]: what is the diffrenece between" cachingConf.enabled" to "cachingConf.useClusterCache"?

File: charts/akeyless-api-gateway/templates/cache.yaml [hunk 7]
```diff
@@ -1,4 +1,4 @@
-{{- if not (eq (include "akeyless-api-gw.adminAccessUidExist" .) "")  }}
+{{- if or (not (eq (include "akeyless-api-gw.adminAccessUidExist" .) "")) ( .Values.cachingConf.useClusterCache) }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/cache.yaml:1]: why the "or" condition with "useClusterCache"? should it be "cachingConf.enabled" ?

File: charts/akeyless-api-gateway/values.yaml (hunk 5 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/values.yaml:356]: no
fixing it.

File: charts/akeyless-api-gateway/README.md (hunk 6 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/README.md:124]: cachingConf.enabled - related for enable caching mode.
cachingConf.useClusterCache - use sepice cache type - cluster cache which is redis

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 8]
```diff
@@ -266,6 +268,15 @@ spec:
                   name: {{ include "akeyless-api-gw.cacheSecretName" . }}
                   key: cache-pass
           {{- end }}
+          {{- if .Values.cachingConf.useClusterCache }}
+            - name: USE_CLUSTER_CACHE
+              value: "true"
+            - name: CACHE_ENCRYPTION_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "akeyless-api-gw.clusterCacheEncryptionKeyExist" . }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:277]: what happens if returns empty?

File: charts/akeyless-api-gateway/templates/cache.yaml (hunk 7 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/templates/cache.yaml:1]: actually for cluster cache we need both:
- `cachingConf.enabled`
- `cachingConf.useClusterCache`

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 8 above)
Comment (idanmantin) [reply charts/akeyless-api-gateway/templates/deployment.yaml:277]: the GW will print an error log, if the EncryptionKey is missing.

File: charts/akeyless-api-gateway/README.md [hunk 9]
```diff
@@ -114,14 +114,15 @@ The following table lists the configurable parameters of the API Gateway chart a
...
+| `cachingConf.clusterCacheEncryptionKeyExistingSecret` | Specifies an existing secret for the cluster cache configuration                                                                                                                                                                      | `false` |
```
Comment (idanmantin) [inline charts/akeyless-api-gateway/README.md:125]: change false to nil, and add:
`, must include:
  # - cluster-cache-encryption-key`

Comment (idanmantin) [reply charts/akeyless-api-gateway/README.md:125]: Done

**Review by niv-ez — APPROVED** [review]

**Review by renanaAkeyless — APPROVED** [review]
